extensions/control-plane/api/management-api/contract-definition-api/src/test/java/org/eclipse/edc/connector/api/management/contractdefinition/ContractDefinitionApiControllerIntegrationTest.java

```diff
@@ -220,15 +221,55 @@ void deleteContractDefinition_notExists() {
                 .statusCode(404);
     }
 
-    private ContractDefinitionRequestDto createDto(String definitionId) {
-        return ContractDefinitionRequestDto.Builder.newInstance()
+    @Test
+    void updateContractDefinition_whenExists(ContractDefinitionStore store) {
+
+        store.accept(createContractDefinition("definitionId"));
+
+        var dto = updateDto();
+
+        baseRequest()
+                .body(dto)
+                .contentType(JSON)
+                .put("/contractdefinitions/definitionId")
+                .then()
+                .statusCode(204);
+
+        assertThat(store.findAll(QuerySpec.max())).hasSize(1);
+    }
+
+    @Test
+    void updateContractDefinition_whenNotExists(ContractDefinitionStore store) {
+
+        var dto = updateDto();
+
+        baseRequest()
+                .body(dto)
+                .contentType(JSON)
+                .put("/contractdefinitions/definitionId")
+                .then()
+                .statusCode(404);
+
+        assertThat(store.findAll(QuerySpec.max())).hasSize(0);
+    }
+
+    private ContractDefinitionCreateDto createDto(String definitionId) {
+        return ContractDefinitionCreateDto.Builder.newInstance()
                 .id(definitionId)
                 .contractPolicyId(UUID.randomUUID().toString())
                 .accessPolicyId(UUID.randomUUID().toString())
                 .criteria(List.of(CriterionDto.Builder.newInstance().operandLeft("left").operator("=").operandRight("right").build()))
                 .build();
     }
 
+    private ContractDefinitionUpdateDto updateDto() {
+        return ContractDefinitionUpdateDto.Builder.newInstance()
+                .contractPolicyId(UUID.randomUUID().toString())
+                .accessPolicyId(UUID.randomUUID().toString())
+                .criteria(List.of(CriterionDto.Builder.newInstance().operandLeft("left").operator("=").operandRight("right").build()))
```

Comment:
i know it may be nit-picking, but updating the criteria to something entirely different might highlight the test case better.